sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormatSuite.scala

```diff
@@ -368,4 +368,14 @@ class BinaryFileFormatSuite extends QueryTest with SharedSQLContext with SQLTest
       assert(caught.getMessage.contains("exceeds the max length allowed"))
     }
   }
+
+  test("SPARK-28030: support chars in file names that require URL encoding") {
```

Comment:
The change seems to impact not just the binary file format, maybe this belongs in one of our root datasource tests. What do you think?

Reply:
it still relies on each data source implementation to recognize `filePath` is actually an URI. I don't see how to test it in our root datasource test. Btw, this PR adds a regression test. So I want to keep the scope minimal.